llvm/lib/Target/PowerPC/PPCISelLowering.cpp

```diff
@@ -775,6 +775,10 @@ PPCTargetLowering::PPCTargetLowering(const PPCTargetMachine &TM,
     setOperationAction(ISD::FMAXNUM_IEEE, MVT::f32, Legal);
     setOperationAction(ISD::FMINNUM_IEEE, MVT::f64, Legal);
     setOperationAction(ISD::FMINNUM_IEEE, MVT::f32, Legal);
+    setOperationAction(ISD::FMAXNUM, MVT::f64, Legal);
+    setOperationAction(ISD::FMAXNUM, MVT::f32, Legal);
+    setOperationAction(ISD::FMINNUM, MVT::f64, Legal);
+    setOperationAction(ISD::FMINNUM, MVT::f32, Legal);
```

Comment:
This is beyond what the title suggests. Do all subtargets have the ignore signaling behavior, and only vsx has the ieee behavior?

Reply:
> This is beyond what the title suggests. 

It's a bootstrap problem:
If we fix `FMAXNUM_IEEE` only, some of test cases will fail.
The problem is that currently `FMINNUM` is expanded to
   FCANONICALIZE + FCANONICALIZE + FMINNUM_IEEE
while FCANONICALIZE is not defined by here: I will submit it in a future patch.

> Do all subtargets have the ignore signaling behavior, and only vsx has the ieee behavior?

It seems that PowerPC doesn't have min/max operation in its scale part.

Reply:
If we fix `expandFMINNUMFMAXNUM` first, some PowerPC test cases will fails with complains that we cannot legalize `FMINNUM_IEEE`.

Reply:
So my plan to break this bootstrap problem is:
1. mark `Legal` to ISD::FMINNUM  for PowerPC/VSX.
   add FMINNUM_IEEE to PowerPC/VSX.
3. add `FCANONICALIZE` support to PowerPC/VSX.
4. Fix expandFMINNUMFMAXNUM to drop `FCANONICALIZE` if `FMINNUM_IEEE` is available.
5. Add test cases of `fmininumnum` to PowerPC/VSX.

Reply:
My concern with the break down here is that it will leave things in an invalid state.  Shouldn't step 1-4 be done together for the compiler behaviour to be valid?  At the minimun, step 4, test casess should be added with the corresponding code change?

Reply:
Let's ask @arsenm. I was asked to split these changes when I was working on other architectures.

Reply:
@arsenm ping